libyul/optimiser/DataFlowAnalyzer.cpp

```diff
@@ -158,15 +159,14 @@ void DataFlowAnalyzer::operator()(Switch& _switch)
 			controlFlowKind(_case.body.statements.back()) == TerminationFinder::ControlFlow::FlowOut
 		)
 		{
-			// TODO maybe we need the symmetric check here because
-			// postState is not always an ancestor of m_storage
 			joinKnowledge(postState);
-			clearValues(assignedVariableNames(_case.body));
+			assignedVariables += assignedVariableNames(_case.body);
 
 			postState = move(m_state);
 		}
 	}
 	m_state = move(postState);
+	clearValues(assignedVariables);
```

Comment:
joinKnowledge only acts on memory and storage, so the assignment `postState = move(m_state)` will just copy over the variable values as they were in the last case.

I'm not yet sure if we also need to clear after joinKnowledge or if it is OK here. We might have to clear above as well.